K2Bridge/Visitors/TermsVisitor.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using K2Bridge.Models.Request.Aggregations;
+
+    /// <content>
+    /// A visitor for the <see cref="TermsAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(TermsAggregation termsAggregation)
+        {
+            Ensure.IsNotNull(termsAggregation, nameof(TermsAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(termsAggregation.Metric, nameof(TermsAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(termsAggregation.FieldName, nameof(TermsAggregation.FieldName));
+
+            termsAggregation.KustoQL = $"{termsAggregation.Metric} by {termsAggregation.FieldAlias} = {termsAggregation.FieldName}";
+
+            if (termsAggregation.SortFieldName == "_key")
+            {
+                // Alphabetical order
+                termsAggregation.KustoQL += $"{KustoQLOperators.CommandSeparator}{KustoQLOperators.OrderBy} {termsAggregation.FieldAlias} {termsAggregation.SortOrder}";
+            }
+            else if (termsAggregation.SortFieldName == "_count")
+            {
+                // Count order
+                termsAggregation.KustoQL += $"{KustoQLOperators.CommandSeparator}{KustoQLOperators.OrderBy} count_ {termsAggregation.SortOrder}";
```

Comment:
is count_ always the column name?

Reply:
I meant in line 31, your wrote order by count_

is that always going to be the column name?

Reply:
At the moment yes, since we currently always query Kusto with at least `summarize count() by ...` for the primary aggregation. Then there is always a column named `count_` in the results. I can use `{BucketColumnNames.Count}` though.

Reply:
Additional details: order by `count_` is used when the terms buckets are ordered by `_count` in the request (it is confusing). If another order metric is specified, then the corresponding aggregation name is used in line 37.